src/components/02_atoms/Widgets/FileUploadWidget.js

```diff
@@ -1,72 +1,199 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
 import FormControl from '@material-ui/core/FormControl';
 import FormLabel from '@material-ui/core/FormLabel';
 import TextField from '@material-ui/core/TextField';
-import Card from '@material-ui/core/Card';
-import CardContent from '@material-ui/core/CardContent';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import Button from '@material-ui/core/Button';
+import DeleteIcon from '@material-ui/icons/Delete';
 
 import WidgetPropTypes from '../../05_pages/NodeForm/WidgetPropTypes';
 import FileUpload from '../FileUpload/FileUpload';
 
-const FileUploadWidget = props => (
-  <Card>
-    <CardContent>
-      <FormControl>
-        <FormLabel component="legend">{props.label}</FormLabel>
-        <FileUpload
-          bundle={props.bundle}
-          entityTypeId={props.entityTypeId}
-          fieldName={props.fieldName}
-          onFileUpload={file => {
-            props.onChange({
-              data: {
-                type: 'file--file',
-                id: file.uuid[0].value,
-              },
-            });
+const Element = styled('div')`
+  > div {
+    width: 100%;
+  }
+
+  legend {
+    margin-bottom: 10px;
+  }
+
+  .remove {
+    margin-left: auto;
+  }
+`;
+
+const Image = styled('div')`
+  > img {
+    max-width: 100px;
+    margin-right: 20px;
+  }
+`;
+
+const FileUploadWidget = ({
+  value,
+  label,
+  bundle,
+  onChange,
+  fieldName,
+  inputProps,
+  entityTypeId,
+  schema: { properties, maxItems },
+}) => {
+  const length = (value && value.data && Object.keys(value.data).length) || 0;
+  const multiple = properties.data.type === 'array'; // If array then allow for multiple uploads
+  const maxItemsCount = maxItems || 1; // maxItems is only set if array, so set to 1 as default
```

Comment:
Isn't the logic? ```multiple ? (maxItems || 100000000000) : 1``` instead?

Reply:
Ah, I assumed maxItems was always set if array. Fixed.

Note to self: Don't assume anything. :P

Reply:
Thank you!